src/fold/base/classes.py

```diff
@@ -266,3 +265,52 @@ def transform(
         self, X: pd.DataFrame, in_sample: bool
     ) -> Tuple[pd.DataFrame, Optional[Artifact]]:
         return pd.concat([X, self.get_function()(X)], axis="columns"), None
+
+
+class EventDataFrame(pd.DataFrame):
+    start: pd.Series
+    end: pd.Series
+    label: pd.Series
+    raw: pd.Series
+
+    def __init__(
+        self,
+        start: pd.DatetimeIndex,
+        end: pd.DatetimeIndex,
+        label: pd.Series,
+        raw: pd.Series,
+    ):
+        super().__init__(data={"start": start, "end": end, "label": label, "raw": raw})
+
+
+@dataclass
+class Extras:
+    events: Optional[EventDataFrame]
+    sample_weights: Optional[pd.Series]
+
+    def __init__(
+        self,
+        events: Optional[EventDataFrame] = None,
+        sample_weights: Optional[pd.Series] = None,
+    ):
+        self.events = events
+        self.sample_weights = sample_weights
+
+    def loc(self, s) -> Extras:
+        return Extras(
```

Comment:
these are just forwarding everything to .loc and .iloc of the underlying dataframes.

pandas has a really weird implementation, and I didn't want to replicate that, they return an indexer object from .loc() and .iloc(), so to use these methods, you need to construct the slice object yourself, can't use the `start:end` shorthand syntax.